### PR TITLE
tools/version.sh: Add the remaining cut to VERSION

### PR DESCRIPTION
**Pull request by microhobby**
## Summary

The idea is to have something like EXTRAVERSION. This is useful for
getting RC tags, development tags and for customizing the version. For
example using the tag `nuttx-10.0.0-RC0` the VERSION will be
`10.0.0-RC0`

## Impact

I really have no idea, I am still assimilating NuttX. So I am leaving this PR as WIP.
This is a suggestion (also to help me understand everything and do something hands-on), so I will wait for feedback from the maintainers.

## Testing

Tested on Linux user-mode:

image

### CODE REVIEW

File: tools/version.sh [hunk 1]
```diff
@@ -81,6 +82,14 @@ if [ -z ${VERSION} ] ; then
 
 fi
 
+if [ -z ${EXTRA} ] ; then
+  EXTRA=`git -C ${WD} tag --sort=v:refname | grep -E "nuttx-[0-9]+\.[0-9]+\.[0-9]+" | tail -1 | cut -d'-' -f3`
```
Comment (xiaoxiang781216) [inline tools/version.sh:86]: maybe, we can extend VERSION to carry the tail part directly.

File: tools/version.sh [hunk 2]
```diff
@@ -25,6 +25,7 @@ USAGE="USAGE: $0 [-d|-h] [-b <build>] [-v <major.minor.patch>] <outfile-path>"
 ADVICE="Try '$0 -h' for more information"
 
 unset VERSION
+unset EXTRA
```
Comment (xiaoxiang781216) [inline tools/version.sh:28]: another way is to utilize -b(BUILD): you can add any custom string as you want.

File: tools/version.sh (hunk 1 above)
Comment (Ouss4) [reply tools/version.sh:86]: Yes, this is better than duplicating the same `git` command.

Comment (microhobby) [reply tools/version.sh]: @xiaoxiang781216 @Ouss4 thanks for the review.
Yeah I agree, and use the remaining of the cut directly in VERSION looks good to me.
I updated the commit

**Comment by acassis** [comment]
Thank you @microhobby  for implementing the suggested approach!

**Pull request merged**